files/fluentd/run.sh

```diff
@@ -86,34 +88,29 @@ if [ $DF_LIMIT -eq 0 ]; then
     echo "ERROR: No disk space is available for file buffer in $FILE_BUFFER_PATH."
     exit 1
 fi
-# Determine final total given the number of outputs we have.
-TOTAL_LIMIT=$(echo ${FILE_BUFFER_LIMIT:-2Gi} | sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc) || :
-if [ $TOTAL_LIMIT -le 0 ]; then
-    echo "ERROR: Invalid file buffer limit ($FILE_BUFFER_LIMIT) is given.  Failed to convert to bytes."
-    exit 1
-fi
 
-TOTAL_LIMIT=$(expr $TOTAL_LIMIT \* $NUM_OUTPUTS) || :
-if [ $DF_LIMIT -lt $TOTAL_LIMIT ]; then
-    echo "WARNING: Available disk space ($DF_LIMIT bytes) is less than the user specified file buffer limit ($FILE_BUFFER_LIMIT times $NUM_OUTPUTS)."
-    TOTAL_LIMIT=$DF_LIMIT
-fi
-
-BUFFER_SIZE_LIMIT=$(echo $BUFFER_SIZE_LIMIT |  sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc)
-BUFFER_SIZE_LIMIT=${BUFFER_SIZE_LIMIT:-16777216}
-
-# TOTAL_BUFFER_SIZE_LIMIT per buffer
-TOTAL_BUFFER_SIZE_LIMIT=$(expr $TOTAL_LIMIT / $NUM_OUTPUTS) || :
-if [ -z $TOTAL_BUFFER_SIZE_LIMIT -o $TOTAL_BUFFER_SIZE_LIMIT -eq 0 ]; then
-    echo "ERROR: Calculated TOTAL_BUFFER_SIZE_LIMIT is 0. TOTAL_LIMIT $TOTAL_LIMIT is too small compared to NUM_OUTPUTS $NUM_OUTPUTS. Please increase FILE_BUFFER_LIMIT $FILE_BUFFER_LIMIT and/or the volume size of $FILE_BUFFER_PATH."
-    exit 1
-fi
-BUFFER_QUEUE_LIMIT=$(expr $TOTAL_BUFFER_SIZE_LIMIT / $BUFFER_SIZE_LIMIT) || :
-if [ -z $BUFFER_QUEUE_LIMIT -o $BUFFER_QUEUE_LIMIT -eq 0 ]; then
-    echo "ERROR: Calculated BUFFER_QUEUE_LIMIT is 0. TOTAL_BUFFER_SIZE_LIMIT $TOTAL_BUFFER_SIZE_LIMIT is too small compared to BUFFER_SIZE_LIMIT $BUFFER_SIZE_LIMIT. Please increase FILE_BUFFER_LIMIT $FILE_BUFFER_LIMIT and/or the volume size of $FILE_BUFFER_PATH."
-    exit 1
-fi
-export BUFFER_QUEUE_LIMIT BUFFER_SIZE_LIMIT
+# Default to 15% of disk which is approximately 18G
+ALLOWED_PERCENT_OF_DISK=${ALLOWED_PERCENT_OF_DISK:-15}
+if [ $ALLOWED_PERCENT_OF_DISK -gt 100 ] || [ $ALLOWED_PERCENT_OF_DISK -le 0 ] ; then
+  echo ALLOWED_PERCENT_OF_DISK is out of the allowed range. Setting to ${ALLOWED_PERCENT_OF_DISK}%
+  ALLOWED_PERCENT_OF_DISK=15
+fi
+# Determine allowed total given the number of outputs we have.
+ALLOWED_DF_LIMIT=$(expr $DF_LIMIT \* $ALLOWED_PERCENT_OF_DISK / 100) || :
+
+# TOTAL_LIMIT_SIZE per buffer
+TOTAL_LIMIT_SIZE=$(expr $ALLOWED_DF_LIMIT / $NUM_OUTPUTS) || :
+echo "Setting each total_size_limit for $NUM_OUTPUTS buffers to $TOTAL_LIMIT_SIZE bytes"
+export TOTAL_LIMIT_SIZE
+
+##
+# Calculate the max number of queued chunks given the size of each chunk
+# and the max allowed space per buffer
+## 
+BUFFER_SIZE_LIMIT=$(echo ${BUFFER_SIZE_LIMIT:-8m} |  sed -e "s/[Kk]/*1024/g;s/[Mm]/*1024*1024/g;s/[Gg]/*1024*1024*1024/g;s/i//g" | bc)
+BUFFER_QUEUE_LIMIT=$(expr $TOTAL_LIMIT_SIZE / $BUFFER_SIZE_LIMIT)
+echo "Setting queued_chunks_limit_size for each buffer to $BUFFER_QUEUE_LIMIT"
+export BUFFER_QUEUE_LIMIT
```

Comment:
formula for queue size comes to: `15% of disk-size/(num_outputs * chunk_size)`
queue size becomes inversely proportional to num_outputs
just a thought, should we keep the queue_size fixed, and calculate the chunk_size ? then chunk size becomes directly proportional to the % of disk size, inverly proportional to num_outputs.

Reply:
I think it has to be calculated and unfortunately is inversely proportional.  The node has a fixed disk available  which we must distribute among the buffer for the outputs.  I believe chunk size is the value we landed on based on what we saw in production but maybe we should consider adjustments in future